### PR TITLE
CLN: lint for ambiguous variable names

### PR DESCRIPTION
**Pull request by jbrockmendel**
One of the codes with a small number of remaining violations.  One-letter names are a PITA in general, are a hassle to track down on the places they are used.

### CODE REVIEW

**Comment by bashtage** [comment]
I can't go for this.  While there are plenty of poorly names variables, some one letters make a lot of sense to people with domain knowledge and improve the code readabililty.  This one falls into unnecessary code churn, at least right now.

File: statsmodels/base/l1_cvxopt.py [hunk 1]
```diff
@@ -169,9 +169,9 @@ def _get_G(k_params):
     """
     The linear inequality constraint matrix.
     """
-    I = np.eye(k_params)
```
Comment (bashtage) [inline statsmodels/base/l1_cvxopt.py]: This one is sill.  I has bad but A and B are ok? I has clear meaning to anyone with linear algebra knowledge, and I_k isn't an improvement.

Comment (bashtage) [reply statsmodels/base/l1_cvxopt.py]: Oh, wrong rule,  But I is still clear, at least with any decent fixed width font.

File: statsmodels/base/tests/test_shrink_pickle.py [hunk 2]
```diff
@@ -57,7 +57,7 @@ def test_remove_data_pickle(self):
 
         #check pickle unpickle works on full results
         #TODO: drop of load save is tested
-        res, l = check_pickle(results._results)
+        res, lbytes = check_pickle(results._results)
```
Comment (bashtage) [inline statsmodels/base/tests/test_shrink_pickle.py]: I'm happier with l changes, although not sure worth it either.  Especially when there isn't a specific meaning to the variable.

File: statsmodels/genmod/tests/test_glm.py [hunk 3]
```diff
@@ -1692,26 +1692,26 @@ def test_wtd_patsy_missing():
 
 class CheckTweedie(object):
     def test_resid(self):
-        l = len(self.res1.resid_response) - 1
+        idx1 = len(self.res1.resid_response) - 1
```
Comment (bashtage) [inline statsmodels/genmod/tests/test_glm.py]: This one changes the flow.  One len names idx1 and the other l2?

**Comment by bashtage** [comment]
If you want to change the lower l to a meaningful name and the noqa the
others, that will let the rule in.

File: statsmodels/nonparametric/kernel_regression.py [hunk 4]
```diff
@@ -875,15 +875,15 @@ def _compute_test_stat(self, Y, X):
 
         m0 = model.fit(data_predict=X1)[0]
         m0 = np.reshape(m0, (n, 1))
-        I = np.zeros((n, 1))
+        I_n = np.zeros((n, 1))
```
Comment (jbrockmendel) [inline statsmodels/nonparametric/kernel_regression.py]: Does `I` have a domain-specific meaning here I'm not aware of?  The other cases are identity matrices.  Unless this has a specific meaning, I might call this `zvec`

File: statsmodels/tsa/interp/denton.py [hunk 5]
```diff
@@ -215,14 +215,14 @@ def dentonm(indicator, benchmark, freq="aq", **kwargs):
 
     # make partitioned matrices
     #TODO: break this out so that we can simplify the linalg?
-    I = zeros((n+m,n+m))
```
Comment (jbrockmendel) [inline statsmodels/tsa/interp/denton.py]: `I` isnt an identity matrix here.  Is it the standard usage in the literature?

Comment (bashtage) [reply statsmodels/tsa/interp/denton.py]: It is not standard but given the other notation in this function I strongly suspect they were following the author.

**Pull request closed without merging**